Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp

```diff
@@ -9171,6 +9172,10 @@ void ByteCodeParser::parse()
     parseCodeBlock();
     linkBlocks(inlineStackEntry.m_unlinkedBlocks, inlineStackEntry.m_blockLinkingTargets);
 
+    m_graph.determineReachability();
+    performLiveCatchVariablePreservationPhase(m_graph);
+    m_graph.clearReachability();
```

Comment:
We should move ` m_graph.determineReachability()` and `m_graph.clearReachability()` inside `performLiveCatchVariablePreservationPhase`, and run them only when `m_graph.m_hasExceptionHandlers` is true.